search_test.go

```diff
@@ -1083,3 +1083,66 @@ func TestDisjunctionQueryIncorrectMin(t *testing.T) {
 			" but got: %v", res.Total)
 	}
 }
+
+func TestBooleanShouldMinPropagation(t *testing.T) {
+	idx, err := New("testidx", NewIndexMapping())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer func() {
+		err = idx.Close()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		err := os.RemoveAll("testidx")
+		if err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	doc1 := map[string]interface{}{
+		"dept":  "finance",
+		"name":  "alvita",
+		"email": "alvita@domain.com",
+	}
+
+	doc2 := map[string]interface{}{
+		"dept":  "dev-ops",
+		"name":  "keelia",
+		"email": "keelia@domain.com",
+	}
+
+	batch := idx.NewBatch()
+
+	if err = batch.Index("doc1", doc1); err != nil {
+		t.Fatal(err)
+	}
+
+	if err = batch.Index("doc2", doc2); err != nil {
+		t.Fatal(err)
+	}
+
+	if err = idx.Batch(batch); err != nil {
+		t.Fatal(err)
+	}
+
+	mq1 := NewMatchQuery("dev-ops")
+	mq1.SetField("dept")
+	mq2 := NewMatchQuery("keelia@domain.com")
```

Comment:
i think this is a bad example, the match query will match both documents, because the way email addresses are analyzed.  you and i know that, but a user reading this test might be confused, or draw incorrect conclusions about why it really works.  it should be simple to change the data/query to avoid this confusion.

Reply:
Fair enough, let me use a better example :)